File: components/js_injection/common/web_message.cc
// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/js_injection/common/web_message.h"

#include <string>

namespace js_injection {

JsWebMessage::JsWebMessage() = default;

JsWebMessage::JsWebMessage(JsWebMessage&&) = default;

JsWebMessage& JsWebMessage::operator=(JsWebMessage&&) = default;

}  // namespace js_injection

File: weblayer/public/java/org/chromium/weblayer/WebMessage.java
// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.weblayer;

import androidx.annotation.NonNull;

/**
 * Used when sending and receiving messages to a page.
 */
class WebMessage {
    private final String mContents;

    /**
     * Creates a message with the specified contents.
     *
     * @param message Contents of the message.
     */
    public WebMessage(@NonNull String message) {
        mContents = message;
    }

    /**
     * Returns the contents of the message.
     *
     * @return The contents of the message.
     */
    public @NonNull String getContents() {
        return mContents;
    }
}
